### ISSUE TITLE
Allow caller to provide host location of kubectl

### ISSUE THREAD

**Issue by markllama** [projectatomic/atomicapp#139]
As an atomicapp user I would like to be able to provide the host location of the kubectl binary
  So that I can use a non-standard or development version of kubectl when testing or
  when running in the Kubernetes Vagrant/Virtualbox build/test environment.

Examples:

````atomic run [mynulecule] --kubectl=/home/user/kubernetes/cluster/kubectl.sh````

or 

````KUBECTL=/home/user/kubernetes/cluster/kubectl.sh atomic run [mynulecule] ````

**Comment by markllama** [comment]
I looked at how the providers are currently structured and realized that I'd need a more generic value for CLI input: ````--provider-cli````. This is appropriate for docker (````docker````), kubernetes (````kubectl````) and OpenShift (````osc````).

atomic run [my atomicapp] --provider-cli=/my/path/to/kubectl [args...]

For the environment variable, each provider will need to set its own variable

- kubernetes - KUBECTL
- docker - DOCKER (??)
- openshift - OSC (??)

Each provider can check the CLI config for 'provider_cli', but I don't see a reason to force ATOMICAPP_PROVIDER_CLI on the whole system yet.

**Comment by markllama** [comment]
It appears that one cannot run an atomic command with args:

````
sudo atomic run docker-registry.usersys.redhat.com/mlamouri/pulp-nulecule --provider-cli=${HOME}/kubernetes/cluster/kubectl.sh
Container 'pulp-nulecule' must be running before executing a command into it.
Execute the following to create the container:
/bin/atomic run --name pulp-nulecule docker-registry.usersys.redhat.com/mlamouri/pulp-nulecule
````

I need to find out how to pass arguments into an atomic run and/or pass in environment variables.

**Comment by vpavlin** [comment]
afaik you cannot pass anything through atomic at the moment. There are some pending issues about this for atomic.

Few notest to this:

* `atomicapp` is mostly run in a container thus we need to check whether user provided path prefixed with `/host` or we need to prepend it ourselves
* we could maybe use `answers.conf` for this - at least until we have support for passing env vars through `atomic`
* I'd like to avoid adding cmdline arguments if possible

**Comment by markllama** [comment]
Re "/host".  The user can't see inside the atomicapp.  The user should
always provide the path from the host perspective.  That means the
atomicapp should always add the /host prefix.

I'd been looking at the answers.conf.  I'm trying to learn how CLI
arguments or environment variables get into the answers file and how the
atomicapp extracts and uses them.

I don't think you can avoid CLI arguments for some things, though this is
really a debug/test issue so it might be acceptable to use alternates
first.  The user is likely to have run-time inputs regardless.

- Mark

On Tue, Jun 2, 2015 at 1:48 AM, Vaclav Pavlin <notifications@github.com>
wrote:

> afaik you cannot pass anything through atomic at the moment. There are
> some pending issues about this for atomic.
>
> Few notest to this:
>
>    - atomicapp is mostly run in a container thus we need to check whether
>    user provided path prefixed with /host or we need to prepend it
>    ourselves
>    - we could maybe use answers.conf for this - at least until we have
>    support for passing env vars through atomic
>    - I'd like to avoid adding cmdline arguments if possible
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectatomic/atomicapp/issues/139#issuecomment-107815870>
> .
>



-- 
----
Mark Lamourine <markllama@gmail.com>
Dad, Hubbie, Software Developer, System Administrator, Road Cyclist

**Comment by goern** [comment]
see also #59

**Comment by aweiteka** [comment]
Resolved by #146
